webworker/index.ts

```diff
@@ -136,6 +138,39 @@ const webWorkerApi: InternalWebWorkerApi = {
     }
     return circuit.getCircuitJson()
   },
+
+  executeWithFs: async ({ fsMap, entrypoint }: ExecuteWithFsOptions) => {
+    // Set up virtual filesystem
+    virtualFs = fsMap
+
+    // Execute the entrypoint file
+    const code = virtualFs[entrypoint]
+    if (!code) {
+      throw new Error(`Entrypoint "${entrypoint}" not found in filesystem map`)
+    }
+
+    // Handle imports by intercepting require/import statements
+    const transformedCode = transformImports(code, virtualFs)
+
+    // Execute the transformed code using the existing execute method
+    await webWorkerApi.execute(transformedCode)
+  },
 }
 
 Comlink.expose(webWorkerApi)
+
+// Helper to transform relative imports to use virtual fs
+function transformImports(code: string, fsMap: Record<string, string>) {
+  // Transform relative imports to use virtual filesystem
+  return code.replace(
+    /import\s+.*\s+from\s+['"]\.\/([^'"]+)['"]/g,
+    (match, path) => {
+      // Add .tsx extension if not present
+      const fullPath = path.endsWith(".tsx") ? path : `${path}.tsx`
+      if (!fsMap[fullPath]) {
+        throw new Error(`Import not found in filesystem: ${fullPath}`)
+      }
+      return `// Virtual import: ${fullPath}\n${fsMap[fullPath]}`
```

Comment:
this technique...

1) doesn't work for `json` files or other extensions
2) changes the exports of the main module, will cause conflicting duplicate exports
3) doesn't handle recursive imports